src/libs/protocols/protos/agent.proto

```diff
@@ -505,3 +507,58 @@ message GetMetricsRequest {}
 message Metrics {
 	string metrics = 1;
 }
+
+message VolumeStatsRequest {
+	// The volume path on the guest outside the container
+	string volume_guest_path = 1;
+}
+
+// This should be kept in sync with CSI NodeGetVolumeStatsResponse (https://github.com/container-storage-interface/spec/blob/master/csi.proto#L1437)
```

Comment:
Thanks for adding this explanatory comment but this is a little worrying. A few questions / thoughts:

- If you've copied the code, are there any copyright/license issues? Which version of the upstream file have you copied (commit hash)?
- Can you put the copied code into a separate csi.proto and `import` it into `agent.proto` to keep it separate?
- What's the plan for keeping this file in sync?
- How are we going to notice it changed and who is going to be responsible for this?
- The top of that file shows, `// Code generated by make; DO NOT EDIT.`
- Ideally, we've vendor in just the definitions we needed but that might require upstream changes to move these types into a separate (and smaller) `.proto` file.
- Which of our [dependencies](https://github.com/kata-containers/kata-containers/blob/main/versions.yaml), if any, will be impacted if the upstream `csi.proto` changes?

Reply:
Thanks! @jodh-intel 
1. The csi spec is under Apache 2.0 license. I assume we don't have any copyright/license issue to copy the code.
2. I have separated the copied ode to its own `csi.proto` following your suggestion. 
3. In term of keeping the file in sync, I think it can be ad-hoc. 
   * Scenario 1: the upstream made a backward-compatible change to the objects we copied (e.g. adding a new optional field), the existing code should still work. We'll need to update the code only if we want to use the new field.
   * Scenario 2: the upstream made a back-incompatible change to the objects (which is unlikely), the code will break at the csi when it returns the volume response. However this failure case always exists.